### DISCUSSION TITLE
How can I modify de doc.text

### DISCUSSION THREAD

**Discussion by LawrenceJavier** [explosion/spaCy#12527] [Help: Coding & Implementations] [answered]
I want to replace some matches found in the text for a list os values but I cant figure out how to rewrite the existing doc with out error. I would prefer not to create a new doc.
Thanks

**Comment by svlandeg** [comment, marked as answer]
Hi! One of spaCy's core principles is that spaCy's processing never modifies the text itself. Any preprocessing step that changes the text should be run before `nlp(text)`. If you share more details & example code, perhaps we can try and find a good solution together?
